syft/pkg/java_metadata.go

```diff
@@ -24,6 +25,7 @@ type JavaMetadata struct {
 	Manifest      *JavaManifest  `mapstructure:"Manifest" json:"manifest,omitempty"`
 	PomProperties *PomProperties `mapstructure:"PomProperties" json:"pomProperties,omitempty" cyclonedx:"-"`
 	PomProject    *PomProject    `mapstructure:"PomProject" json:"pomProject,omitempty"`
+	Digest        *file.Digest   `hash:"ignore" json:"digest,omitempty"`
```

Comment:
nit: (opinion) a slightly more accurate name for this field is `archiveDigest`, so that it indicates what it is a digest of. what do you think?